lib/Sema/CSStep.cpp

```diff
@@ -356,34 +356,43 @@ StepResult ComponentStep::take(bool prevFailed) {
   });
 
   auto *disjunction = CS.selectDisjunction();
-  auto *conjunction = CS.selectConjunction();
 
   if (CS.isDebugMode()) {
-    PrintOptions PO;
-    PO.PrintTypesForDebugging = true;
-
-    auto &log = getDebugLogger();
     if (!potentialBindings.empty()) {
+      auto &log = getDebugLogger();
       log << "(Potential Binding(s): " << '\n';
       log << potentialBindings;
     }
-    log.indent(CS.solverState->getCurrentIndent());
 
-    if (disjunction) {
-      log.indent(2);
-      log << "Disjunction(s) = [";
+    SmallVector<Constraint *, 4> disjunctions;
+    CS.collectDisjunctions(disjunctions);
+    std::vector<std::string> choicesWithBindOverload;
+    for (const auto &disjunction : disjunctions) {
+      PrintOptions PO;
+      PO.PrintTypesForDebugging = true;
+
+      bool isBindOverload = false;
       auto constraints = disjunction->getNestedConstraints();
-      log << constraints[0]->getFirstType()->getString(PO);
-      log << "]";
+      for (const auto &choice : constraints) {
+        if (choice->getKind() == ConstraintKind::BindOverload)
```

Comment:
I realize that I should have explained better - We actually expect that if first choice is `BindOverload` all choices are so you don't have to loop here just check `constraints[0]->getKind() == ConstraintKind::BindOverload`